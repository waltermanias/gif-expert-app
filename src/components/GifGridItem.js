import React from "react";
import PropTypes from "prop-types";

const GifGridItem = ({ title, id, url }) => {
  return (
    <div className="card animate__animated animate__bounce animate__delay-2s">
      <img src={url} alt={"imagen"} />
      <p>{title}</p>
    </div>
  );
};

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};

export default GifGridItem;
