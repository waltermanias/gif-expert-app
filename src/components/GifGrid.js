import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
import GifGridItem from "./GifGridItem";

export const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);

  return (
    <>
      <h3>{category}</h3>
      {loading ? "Cargando" : "Data cargada"}
      <div className="card-grid">
        {images.map((image) => {
          return <GifGridItem key={image.id} {...image} />;
        })}
      </div>
    </>
  );
};
