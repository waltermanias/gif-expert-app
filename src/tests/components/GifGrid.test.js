import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from "../../hooks/useFetchGifs";

jest.mock("../../hooks/useFetchGifs");

describe("GifGrid tests", () => {
  it("should render the component", () => {
    useFetchGifs.mockReturnValue({ data: [], loading: true });
    const wrapper = shallow(<GifGrid category="One Punch" />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render items", () => {
    const fakeData = {
      data: [
        { id: "ABC", title: "test", url: "http://localhost:3000/image.jpg" },
      ],
      loading: false,
    };
    useFetchGifs.mockReturnValue(fakeData);
    const wrapper = shallow(<GifGrid category="One Punch" />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("p").exists()).toBe(false);
    expect(wrapper.find("GifGridItem").length).toBe(fakeData.data.length);
  });
});
