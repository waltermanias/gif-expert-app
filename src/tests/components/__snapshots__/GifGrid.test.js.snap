// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid tests should render items 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  Data cargada
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="test"
      url="http://localhost:3000/image.jpg"
    />
  </div>
</Fragment>
`;

exports[`GifGrid tests should render the component 1`] = `
<Fragment>
  <h3>
    One Punch
  </h3>
  Cargando
  <div
    className="card-grid"
  />
</Fragment>
`;
